test/gtest/ucp/ucp_test.cc

```diff
@@ -253,10 +286,10 @@ ucs_status_t ucp_test::request_process(void *req, int worker_index, bool wait)
     ucs_status_t status;
     if (wait) {
         ucs_time_t deadline = ucs::get_deadline();
-        do {
-            progress(worker_index);
-            status = ucp_request_check_status(req);
-        } while ((status == UCS_INPROGRESS) && (ucs_get_time() < deadline));
+        while (((status = ucp_request_check_status(req)) ==
```

Comment:
maybe remove status variable all together? it would simplify the code
```
ucs_time_t deadline = ucs::get_deadline();
while (wait &&
       (ucp_request_check_status(req) == UCS_INPROGRESS) &&
       (ucs_get_time() < deadline)) {
    check_events(entities(), wakeup, worker_index);
}

return ucp_request_check_status(req);
```

Reply:
done